plan/predicate_push_down.go

```diff
@@ -564,16 +566,78 @@ func (p *Union) PredicatePushDown(predicates []expression.Expression) (ret []exp
 }
 
 // PredicatePushDown implements LogicalPlan PredicatePushDown interface.
-func (p *Aggregation) PredicatePushDown(predicates []expression.Expression) ([]expression.Expression, LogicalPlan, error) {
+func (p *Aggregation) PredicatePushDown(predicates []expression.Expression) (ret []expression.Expression, retPlan LogicalPlan, err error) {
 	// TODO: implement aggregation push down.
+	retPlan = p
+	colsOfGroupBy := p.GroupByItems
+	var aggrID, _ = strconv.Atoi(strings.Split(p.GetID(), "_")[1]) // form of Aggregation ID: aggregation_x
+	var colIDsInGroupBy []int
+	var exprs []expression.Expression
+	for id, aggrFun := range p.AggFuncs {
+		exprs = append(exprs, aggrFun.GetArgs()[0])
+		for _, col := range colsOfGroupBy {
+			if aggrFun.GetName() == ast.AggFuncFirstRow && col.Equal(aggrFun.GetArgs()[0]) {
+				colIDsInGroupBy = append(colIDsInGroupBy, id)
+			}
+		}
+	}
 	var condsToPush []expression.Expression
 	for _, cond := range predicates {
-		if _, ok := cond.(*expression.Constant); ok {
+		switch v := cond.(type) {
+		case *expression.Constant:
 			condsToPush = append(condsToPush, cond)
+			ret = append(ret, cond)
+		case *expression.ScalarFunction:
+			if ok, col := scalarFunctionByOneCol(v); ok && col.IsAggOrSubq && colNameIsGroupByItem(col.String(), aggrID, colIDsInGroupBy) {
```

Comment:
If you want to check if the column is group by item, you can get the column's index id directly by p.schema.GetIndex(col).